src/test/Events/EventTemplates.test.tsx

```diff
@@ -128,114 +130,89 @@ describe('<EventTemplates />', () => {
   });
 
-  it('removes a recording after receiving a notification', () => {
-    render(
-      <ServiceContext.Provider value={defaultServices}>
-        <EventTemplates />
-      </ServiceContext.Provider>
-    );
+  it('removes a recording after receiving a notification', async () => {
+    renderWithServiceContext(<EventTemplates />);
+
     expect(screen.queryByText('anotherEventTemplate')).not.toBeInTheDocument();
   });
 
-  it('displays the column header fields', () => {
-    render(
-      <ServiceContext.Provider value={defaultServices}>
-        <EventTemplates />
-      </ServiceContext.Provider>
-    );
+  it('displays the column header fields', async () => {
+    renderWithServiceContext(<EventTemplates />);
+
     expect(screen.getByText('Name')).toBeInTheDocument();
     expect(screen.getByText('Description')).toBeInTheDocument();
     expect(screen.getByText('Provider')).toBeInTheDocument();
     expect(screen.getByText('Type')).toBeInTheDocument();
   });
 
-  it('shows a popup when uploading', () => {
-    render(
-      <ServiceContext.Provider value={defaultServices}>
-        <EventTemplates />
-      </ServiceContext.Provider>
-    );
+  it('shows a popup when uploading', async () => {
+    const { user } = renderWithServiceContext(<EventTemplates />);
+
     expect(screen.queryByLabelText('Create Custom Event Template')).not.toBeInTheDocument();
 
     const buttons = screen.getAllByRole('button');
     const uploadButton = buttons[0];
-    userEvent.click(uploadButton);
+    await user.click(uploadButton);
 
     expect(screen.getByLabelText('Create Custom Event Template'));
   });
 
-  it('downloads an event template when Download is clicked on template action bar', () => {
-    render(
-      <ServiceContext.Provider value={defaultServices}>
-        <EventTemplates />
-      </ServiceContext.Provider>
-    );
+  it('downloads an event template when Download is clicked on template action bar', async () => {
+    const { user } = renderWithServiceContext(<EventTemplates />);
 
-    userEvent.click(screen.getByLabelText('Actions'));
-    userEvent.click(screen.getByText('Download'));
+    await user.click(screen.getByLabelText('Actions'));
+    await user.click(screen.getByText('Download'));
 
     const downloadRequestSpy = jest.spyOn(defaultServices.api, 'downloadTemplate');
 
     expect(downloadRequestSpy).toHaveBeenCalledTimes(1);
     expect(downloadRequestSpy).toBeCalledWith(mockCustomEventTemplate);
   });
 
-  it('shows a popup when Delete is clicked and then deletes the template after clicking confirmation Delete', () => {
-    render(
-      <ServiceContext.Provider value={defaultServices}>
-        <EventTemplates />
-      </ServiceContext.Provider>
-    );
+  it('shows a popup when Delete is clicked and then deletes the template after clicking confirmation Delete', async () => {
+    const { user } = renderWithServiceContext(<EventTemplates />);
```

Comment:
So, the way to do userEvent tests now, are to get the `user` obj from the render function every time instead of explicitly calling `userEvent.click`, etc.?

Reply:
Yess exactly! This is meant to call `userEvent.setup` on every test.

Just to clarify: this render function is custom that we wrap around the testing-library render function.